src/libpmemset/part.c

```diff
@@ -159,35 +160,56 @@ pmemset_part_map_new(struct pmemset_part_map **part_map,
 	struct pmem2_source *pmem2_src;
 	pmem2_src = pmemset_file_get_pmem2_source(part->file);
 
-	/* mapping with a hint address next to previous mapped part */
-	void *contiguous_addr = (char *)previous_part.addr + previous_part.size;
 	size_t part_size = part->length;
 	if (part_size == 0)
 		pmem2_source_size(pmem2_src, &part_size);
 
 	struct pmem2_vm_reservation *pmem2_reserv;
-	pmem2_vm_reservation_new(&pmem2_reserv, contiguous_addr, part_size);
+	size_t offset = 0;
 
-	pmem2_config_set_vm_reservation(pmem2_cfg, pmem2_reserv, 0);
+	/* coalesce part mappings into single vm reservation */
+	if (part_coalescing && prev_pmap) {
+		pmem2_reserv = prev_pmap->pmem2_reserv;
+		offset = pmem2_vm_reservation_get_size(pmem2_reserv);
+		ret = pmem2_vm_reservation_extend(pmem2_reserv, part_size);
+	} else
```

Comment:
use brackets uniformly in a series of statements. `if {} else {}` or `if ... else ...;`, never a mix.